addons/base_automation/models/base_automation.py

```diff
@@ -30,25 +30,108 @@
     False: 0,
 }
 
+ON_CREATE = 'on_create'  # deprecated, use ON_CREATE_OR_WRITE instead
+ON_CREATE_OR_WRITE = 'on_create_or_write'
+ON_STAGE_SET = 'on_stage_set'
+ON_USER_SET = 'on_user_set'
+ON_TAG_SET = 'on_tag_set'
+ON_STATE_SET = 'on_state_set'
+ON_PRIORITY_SET = 'on_priority_set'
+CREATE_TRIGGERS = [
+    ON_CREATE,
+    ON_CREATE_OR_WRITE,
+    ON_STAGE_SET,
+    ON_USER_SET,
+    ON_TAG_SET,
+    ON_STATE_SET,
+    ON_PRIORITY_SET,
+]
+
+ON_WRITE = 'on_write'  # deprecated, use ON_CREATE_OR_WRITE instead
+ON_ARCHIVE = 'on_archive'
+ON_UNARCHIVE = 'on_unarchive'
+WRITE_TRIGGERS = [
+    trigger for trigger in CREATE_TRIGGERS if trigger != ON_CREATE
+] + [ON_WRITE, ON_ARCHIVE, ON_UNARCHIVE]
+
+UPDATE_TRIGGERS = list(set(WRITE_TRIGGERS + CREATE_TRIGGERS))
+
+ON_UNLINK = 'on_unlink'
+ON_CHANGE = 'on_change'
+
+ON_TIME = 'on_time'
+ON_TIME_CREATED = 'on_time_created'
+ON_TIME_UPDATED = 'on_time_updated'
+TIME_TRIGGERS = [
+    ON_TIME,
+    ON_TIME_CREATED,
+    ON_TIME_UPDATED,
+]
+
 
 class BaseAutomation(models.Model):
     _name = 'base.automation'
-    _description = 'Automated Action'
-    _order = 'sequence'
+    _description = 'Automation Rule'
 
-    action_server_id = fields.Many2one(
-        'ir.actions.server', 'Server Actions',
-        domain="[('model_id', '=', model_id)]",
-        delegate=True, required=True, ondelete='restrict')
+    name = fields.Char(string="Automation Rule Name", required=True, translate=True)
+    model_id = fields.Many2one(
+        "ir.model", string="Model", required=True, ondelete="cascade", help="Model on which the automation rule runs."
+    )
+    model_name = fields.Char(related="model_id.model", string="Model Name", readonly=True, store=True)
```

Comment:
IDK
This field was copied from ir.actions.server
It is stored since https://github.com/odoo/odoo/commit/bcc0212bfb221976864bbf40038a9093232ce029#diff-5fa54cc9c3a92f43538da6b7d6f88c2f7e93e66fc42a7c8ee31039d9110f04e9